app/components/Form/ProjectManagerForm.tsx

```diff
@@ -38,27 +69,168 @@ const ProjecManagerForm: React.ForwardRefRenderFunction<any, Props> = (
         }
       }
     `,
-    props.allUsers
+    props.query
   );
 
-  const schema: JSONSchema7 = useMemo(() => {
-    const initialSchema = projectManagerSchema;
-
-    initialSchema.properties.cifUserId = {
-      ...initialSchema.properties.cifUserId,
+  // Dynamically build the schema from the list of cif_users
+  const managerSchema = useMemo(() => {
+    const schema = projectManagerSchema;
+    schema.properties.cifUserId = {
+      ...schema.properties.cifUserId,
       anyOf: allCifUsers.edges.map(({ node }) => {
         return {
           type: "number",
           title: `${node.firstName} ${node.lastName}`,
           enum: [node.rowId],
           value: node.rowId,
-        };
+        } as JSONSchema7Definition;
       }),
     };
-    return initialSchema as JSONSchema7;
+
+    return schema as JSONSchema7;
   }, [allCifUsers]);
 
-  return <FormBase {...props} ref={ref} schema={schema} uiSchema={uiSchema} />;
+  // Add a manager to the project revision
+  const [applyAddManagerToRevision] = useAddManagerToRevisionMutation();
+  const addManager = (data: any) => {
+    applyAddManagerToRevision({
+      variables: {
+        projectRevision: projectRevision.id,
+        projectRevisionId: projectRevision.rowId,
+        newFormData: data,
+      },
+    });
+  };
+
+  // Delete a manager from the project revision
+  const [discardFormChange] = useDeleteManagerFromRevisionMutation();
+  const deleteManager = (id: string) => {
+    discardFormChange({
```

Comment:
The action here needs to depend on the `form_change` `operation`:

- If `operation === "CREATE"`, then you can discard it, as the manager was added in the same revision
- If `operation === "UPDATE"`, then the `form_change.operation`  should be updated to `ARCHIVE`